### PR TITLE
docs(macros): Improve parameterized macros docs

### PR DESCRIPTION
**Pull request by minusfive**
This PR improves Parameterized Macros documentation.

Screenshot 2023-08-02 at 11 13 44 AM

Screenshot 2023-08-02 at 11 14 04 AM

Screenshot 2023-08-02 at 11 16 18 AM

Screenshot 2023-08-02 at 11 17 05 AM

### CODE REVIEW

File: docs/docs/behaviors/macros.md [hunk 1]
```diff
@@ -185,6 +145,70 @@ Macros use an internal queue to invoke each behavior in the bindings list when t
 
 To prevent issues with longer macros, you can change the size of this queue via the `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE` setting in your configuration, [typically through your `.conf` file](../config/index.md). For example, `CONFIG_ZMK_BEHAVIORS_QUEUE_SIZE=512` would allow your macro to type about 256 characters.
 
+## Parameterized Macros
+
+Macros can also be "parameterized", allowing them to be bound in your keymap with unique values passed into them, e.g.:
+
+```
+    raise_layer {
+        bindings = <&my_parameterized_macro A>
```
Comment (caksoylar) [inline docs/docs/behaviors/macros.md:154]:
```suggestion
        bindings = <&my_one_param_macro A>;
```
More consistent with the following section.

File: docs/docs/behaviors/macros.md [hunk 2]
```diff
@@ -185,6 +145,70 @@ Macros use an internal queue to invoke each behavior in the bindings list when t
...
+    };
+```
+
+### Defining Parameterized Macros
+
+Parameterized macros must be defined using specific values for the `compatible` and `#binding-cells` properties, depending on how many parameters they require (max `2`):
```
Comment (caksoylar) [inline docs/docs/behaviors/macros.md:160]:
```suggestion
Parameterized macros must be defined using specific values for the `compatible` and `#binding-cells` properties, depending on how many parameters they require (up to a maximum of two):
```

File: docs/docs/behaviors/macros.md [hunk 3]
```diff
@@ -185,6 +145,70 @@ Macros use an internal queue to invoke each behavior in the bindings list when t
...
+            #binding-cells = <2>; // Must be 2
+            bindings = /* ... */;
+        };
+    };
+};
+```
+
+### Parameters, Bindings and Controls
+
+There are special macro controls which must be used in order to forward received parameters to the macro's `bindings`. These controls are "one shot" and will determine how received parameters are used on the very next [non-macro control] behavior in the macro's `bindings` list.
```
Comment (caksoylar) [inline docs/docs/behaviors/macros.md:194]:
```suggestion
There are special macro controls which must be used in order to forward received parameters to the macro's `bindings`. These controls are "one shot" and will determine how received parameters are used on the very next (non-macro control) behavior in the macro's `bindings` list.
```

File: docs/docs/behaviors/macros.md [hunk 4]
```diff
@@ -198,12 +222,35 @@ To achieve this, a combination of a 0ms wait time and splitting the press and re
...
+    bindings
+        = <&macro_param_1to1>,
+        <&macro_press &mo MACRO_PLACEHOLDER>,
+        <&macro_param_2to1>,
+        <&macro_press &kp MACRO_PLACEHOLDER>,
+        <&macro_pause_for_release>,
+        <&macro_param_2to1>,
+        <&macro_release &kp MACRO_PLACEHOLDER>,
+        <&macro_param_1to1>,
+        <&macro_release &mo MACRO_PLACEHOLDER>;
```
Comment (caksoylar) [inline docs/docs/behaviors/macros.md:252]:
```suggestion
        = <&macro_param_1to1>
        , <&macro_press &mo MACRO_PLACEHOLDER>
        , <&macro_param_2to1>
        , <&macro_press &kp MACRO_PLACEHOLDER>
        , <&macro_pause_for_release>
        , <&macro_param_2to1>
        , <&macro_release &kp MACRO_PLACEHOLDER>
        , <&macro_param_1to1>
        , <&macro_release &mo MACRO_PLACEHOLDER>
        ;
```
Style consistency with the rest of the macro docs

File: docs/docs/config/behaviors.md [hunk 5]
```diff
@@ -131,26 +131,31 @@ See the [macro behavior](../behaviors/macros.md) documentation for more details
...
+| `&macro_wait_time TIME`    | Changes the time to wait (in milliseconds) before triggering the next behavior.                                                             |
+| `&macro_tap_time TIME`     | Changes the time to wait (in milliseconds) between the press and release events of a tapped behavior.                                       |
+| `&macro_param_1to1`        | Forward the first parameter received by the macro to the first parameter of the next [non-macro control] behavior in the `bindings` list.   |
+| `&macro_param_1to2`        | Forward the first parameter received by the macro to the second parameter of the next [non-macro control] behavior in the `bindings` list.  |
+| `&macro_param_2to1`        | Forward the second parameter received by the macro to the first parameter of the next [non-macro control] behavior in the `bindings` list.  |
+| `&macro_param_2to2`        | Forward the second parameter received by the macro to the second parameter of the next [non-macro control] behavior in the `bindings` list. |
```
Comment (caksoylar) [inline docs/docs/config/behaviors.md:158]:
```suggestion
| `&macro_param_1to1`        | Forward the first parameter received by the macro to the first parameter of the next (non-macro control) behavior   |
| `&macro_param_1to2`        | Forward the first parameter received by the macro to the second parameter of the next (non-macro control) behavior  |
| `&macro_param_2to1`        | Forward the second parameter received by the macro to the first parameter of the next (non-macro control) behavior  |
| `&macro_param_2to2`        | Forward the second parameter received by the macro to the second parameter of the next (non-macro control) behavior |
```
We can probably be a bit more concise here, will require a `prettier` pass.

**Review by caksoylar — COMMENTED** [review]
Appreciate all the improvements, especially updating the LM example. Mostly minor improvement suggestions above.

Since you are touching these bits, would you also mind adding a note to the "Convenience C Macro" section to specify it defines a zero-parameter macro?

File: docs/docs/behaviors/macros.md (hunk 1 above)
Comment (minusfive) [reply docs/docs/behaviors/macros.md]: Done

File: docs/docs/behaviors/macros.md (hunk 2 above)
Comment (minusfive) [reply docs/docs/behaviors/macros.md]: Done

File: docs/docs/behaviors/macros.md (hunk 3 above)
Comment (minusfive) [reply docs/docs/behaviors/macros.md]: Done

File: docs/docs/behaviors/macros.md (hunk 4 above)
Comment (minusfive) [reply docs/docs/behaviors/macros.md]: Done

**Comment by minusfive** [comment]
@caksoylar addressed all your comments and added the C macro bit. Updated screenshots on main PR description.

**Review by caksoylar — APPROVED** [review]
Thanks again!

**Pull request merged**